src/com/facebook/buck/cxx/HeaderSymlinkTreeWithHeaderMap.java

```diff
@@ -72,10 +120,44 @@ public SourcePath getSourcePathToOutput() {
       // aligning in order to get this to work. May we find peace in another life.
       headerMapEntries.put(key, buckOut.relativize(getRoot().resolve(key)));
     }
-    return ImmutableList.<Step>builder()
+    ImmutableList.Builder<Step> builder = ImmutableList.<Step>builder()
         .addAll(super.getBuildSteps(context, buildableContext))
-        .add(new HeaderMapStep(getProjectFilesystem(), headerMapPath, headerMapEntries.build()))
-        .build();
+        .add(new HeaderMapStep(getProjectFilesystem(), headerMapPath, headerMapEntries.build()));
+
+    if (shouldCreateModule) {
```

Comment:
Always create module map doesn't hurt, except that if the target name is not in a good format for module name, then we have to normalize it (see below)